ngrinder-groovy/src/main/java/net/grinder/scriptengine/groovy/junit/GrinderRunner.java

```diff
@@ -300,8 +295,8 @@ protected Statement withBeforeThread(Statement statement) {
 	protected void registerRunNotifierListener(RunNotifier notifier) {
 		notifier.addFirstListener(new RunListener() {
 			@Override
-			public void testStarted(Description description) {
-
+			public void testFinished(Description description) throws Exception {

```

Comment:
@imbyungjun 

This hook is `Called when an atomic test has finished`
If test script has `@Test` function more than 2, Isn't the connection closed in every function call? and user uses a `Jython` script. This hook will not be called.

I think connection reset logic should be in `ThreadLifeCycleListener.beginRun()` or `ThreadLifeCycleListener.endRun()`

Reply:
@imbyungjun 

Check `GrinderThread.run()`, The hook is called before and after `workerRunnable.run()`

```
     int currentRun;

...

      for (currentRun = 0;
           numberOfRuns == 0 || currentRun < numberOfRuns;
           currentRun++) {

        m_context.setCurrentRunNumber(currentRun);

        m_context.fireBeginRunEvent();

        try {
          workerRunnable.run();
        }
        catch (ScriptExecutionException e) {
          final Throwable cause = e.getCause();

          if (cause instanceof ShutdownException ||
              cause instanceof Sleeper.ShutdownException) {
            m_logger.info(m_context.getLogMarker(), "shut down");
            break;
          }

          m_logger.error(m_context.getLogMarker(),
                         "Aborted run: " + e.getShortMessage(),
                         e);
        }

        m_context.fireEndRunEvent();
      }

...

      m_context.setCurrentRunNumber(-1);
```

Reply:
@imbyungjun 

Old `HTTPRequest` seems to reset the connection in [here](https://github.com/naver/ngrinder/blob/feature/support-http2/ngrinder-runtime/src/main/java/net/grinder/plugin/http/HTTPPluginThreadState.java#L108)

Reply:
@donggyu04 
Thank you for letting me know it. Reset on `beginRun`(as it was) would be great.